File: LogicalThinkingPatterns/AlphaRampTrianglePattern/AlphaRampTriangle.py
class AlphaRampTriangle:
    def printPattern(self, n):
        ch = 'A'
        for i in range(0, n):
            for j in range(0, i+1):
                print(ch+" ", end = "")
            print()
            ch = chr(ord('A')+i+1)
    
    def run(self):
        n = int(input("Enter the number of rows: "))
        self.printPattern(n)

alphaRampTriangle = AlphaRampTriangle()
alphaRampTriangle.run()

File: LogicalThinkingPatterns/AlphaRampTrianglePattern/AlphaRampTriangle.ts
class AlphaRampTriangle {
  printPattern(n: number): void {
    let ch: number = 65;
    for (let i = 0; i < n; i++) {
      let pattern: string = "";
      for (let j = 0; j <= i; j++) {
        pattern += String.fromCharCode(ch) + " ";
      }
      console.log(pattern);
      ch++;
    }
  }
}

const alphaRampTriangle = new AlphaRampTriangle();
alphaRampTriangle.printPattern(5);
